comment.php

```diff
@@ -107,7 +107,7 @@
 	{
 		$pid 				= intval(varset($_POST['pid'], 0)); // ID of the specific comment being edited (nested comments - replies)
 		$row 				= array();
-		$authName           = filter_var($_POST['author_name'],FILTER_SANITIZE_STRING);
+		$authName           = e107::getParser()->filter($_POST['author_name'], 'str');
```

Comment:
@Deltik Maybe I missed something, doesn't this do exactly the same as the code you replaced?

```
public function filter($text, $type = 'str', $validate = false)

...



if($validate === false)
				{
					$filterTypes = array(
						'int'   => FILTER_SANITIZE_NUMBER_INT,
						'str'   => FILTER_SANITIZE_STRING, // no html.
						'email' => FILTER_SANITIZE_EMAIL,
						'url'   => FILTER_SANITIZE_URL,
						'enc'   => FILTER_SANITIZE_ENCODED
					);
				}
				```

Reply:
`e_parse::filter()` has been reworked not to use `filter_var(…, FILTER_SANITIZE_STRING)`.  ([Source](https://github.com/e107inc/e107/pull/4554/files#diff-dc57abd51013ddd4e6bad33d34ac2dae0d2e5302900c00ed69dd4dee3556d939R4855))

According to [the documentation of the deprecation of `FILTER_SANITIZE_STRING`](https://wiki.php.net/rfc/deprecations_php_8_1#filter_sanitize_string), the usage of the filter is unclear.  I looked at e107's usages of that filter and found that some of them expected [`strip_tags()`](https://www.php.net/manual/en/function.strip-tags.php) behavior while others thought that the filter would make the provided string safe to quote in an SQL query.  As there is no way for a filter to concatenate arbitrary strings safely for SQL queries, I decided to make `e_parse::filter(…, 'str', false)` just mean `strip_tags()`.

I replaced the confounded usages of `filter_var(…, FILTER_SANITIZE_STRING)` with guesses for what the usages' intentions were:
* Preventing SQL injection – [Example](https://github.com/e107inc/e107/pull/4554/files#diff-061a207c81a13f011e0c692099bfffd7bad34c9650f804f9968b022d9691b5cdR1389-R1402)
* Remove HTML tags – [Example](https://github.com/e107inc/e107/pull/4554/files#diff-068ed6f9c1bcc16dcb2cec1aa9ebccec2bcd68400ae947c9b53465a885a2d3c6R202)
* Kinda hide quotation marks for use in SQL – [Example](https://github.com/e107inc/e107/pull/4554/files#diff-dc57abd51013ddd4e6bad33d34ac2dae0d2e5302900c00ed69dd4dee3556d939R511)
* No apparent reason at all – [Example](https://github.com/e107inc/e107/pull/4554/files#diff-19e3c84c1b3ce87245588ae6ee7791fe3b94c98a8f955ea3e89f9f5865144ca0L268)